File: hue.php
<?php

/**
 * 
 */
class Hue
{
    protected $url;
    protected $curl;

    public function __construct($bridgeIp, $userName, $ligthNum)
    {
        $this->url = sprintf('http://%s/api/%s/lights/%d/state', $bridgeIp, $userName, $ligthNum);
        $this->curl = curl_init($this->url);
    }

    public function sendRequest($params, $method = 'GET')
    {
        if (strtolower($method) === 'post') {
            curl_setopt($this->curl, CURLOPT_POST, true);
            //curl_setopt($this->curl, CURLOPT_POSTFIELDS, $params);
        }

        if (strtolower($method) === 'put') {
            curl_setopt($this->curl, CURLOPT_CUSTOMREQUEST, "PUT");
            curl_setopt($this->curl, CURLOPT_RETURNTRANSFER, true);
            curl_setopt($this->curl, CURLOPT_HEADER, false);
            // curl_setopt($this->curl, CURLOPT_POSTFIELDS, http_build_query($params));
        }

        curl_setopt($this->curl, CURLOPT_POSTFIELDS, json_encode($params));

        curl_setopt($this->curl, CURLOPT_FOLLOWLOCATION, true);
        return curl_exec($this->curl);
    }

    public function turnOn($hue, $bri = 255, $sat = 255)
    {
        return $this->sendRequest(['on' => true, 'hue' => $hue, 'bri' => $bri, 'sat' => $sat], 'put');
    }

    public function setSate($hue, $bri = 255, $sat = 255)
    {
        return $this->sendRequest(['hue' => $hue, 'bri' => $bri, 'sat' => $sat], 'put');
    }

    public function turnOff()
    {
        return $this->sendRequest(['on' => false]);
    }

}

$colors = [
    'red' => 0,
    'yellow' => 12750,
    'green' => 25500,
    'blue' => 46920,
    'pink' => 56100,
    'red' => 65279,
];

$userName = "r47ty-ubSNwD10DwmbaF-lxLx2xnzeQU42xRsA2u";
$ligthNum = 4;

$hueApi = new Hue('172.21.28.114', $userName, $ligthNum);
$hueApi->turnOn(0);
foreach ($colors as $color => $hue) {

    for ($sat = 0; $sat < 255; $sat += 30) {
        for ($bri = 0; $bri < 255; $bri += 30) {
            $result = $hueApi->turnOn($hue, $bri, $sat);
            $result = json_decode($result);
            if(!empty($result->error)) {
                var_dump($result);
                break 3;
            }
            sleep(0.5);
        }
        sleep(1);
    }
    sleep(1);
}
$hueApi->turnOff(0);

File: Hue.py
import requests, json


class Hue:
    url = False

    def __init__(self, bridgeIp, userName, ligthNum):
        self.url = "http://" + bridgeIp + "/api/" + userName + "/lights/" + str(ligthNum) + "/state"

    def sendRequest(self, data={}, method="GET"):
        method = method.lower();
        print(data);

        if method == 'get':
            response = requests.get(self.url, data)

        if method == 'post':
            response = requests.post(self.url, data)

        if method == 'put':
            response = requests.put(self.url, data=json.dumps(data))

        if method == 'delete':
            response = requests.delete(self.url)

        return self.handleResponse(response)

    def handleResponse(self, response):
        jsonResponse = json.loads(response.text)
        print(jsonResponse)
        if response.status_code == 200:
            return jsonResponse
        else:
            return response

    def turnOn(self, hue, bri=255, sat=255):
        return self.sendRequest({"on": True, "hue": hue, "bri": bri, "sat": sat}, 'PUT')

    def setSate(self, hue, bri=255, sat=255):
        return self.sendRequest({"hue": hue, "bri": bri, "sat": sat}, 'PUT')

    def turnOff(self):
        return self.sendRequest({"on": False}, 'put')
